amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/async/AsyncApiDocumentParser.scala

```diff
@@ -51,15 +51,16 @@ abstract class AsyncApiDocumentParser(root: Root)(implicit val ctx: AsyncWebApiC
     val references = AsycnReferencesParser(root.references).parse()
     parseDeclarations(map)
     val declarationKeys = ctx.getDeclarationKeys
-    if (declarationKeys.nonEmpty) document.add(DeclarationKeys(declarationKeys))
 
     val api = parseApi(map).add(SourceVendor(ctx.vendor))
     document
       .withEncodes(api)
       .adopted(root.location)
 
     val declarable = ctx.declarations.declarables()
-    if (declarable.nonEmpty) document.withDeclares(declarable)
+    val ann        = Annotations(DeclarationKeys(declarationKeys))
```

Comment:
[OOS] - Not sure if using the ctx to accumulate `DeclarationKeys` is the best approach. Isn't this polluting the already heavily polluted WebApiContext? Besides, DeclarationKeys are only added in DocumentParsers.

Reply:
Moved to DeclarationKeysCollector